rust/parquet/src/arrow/schema.rs

```diff
@@ -42,12 +42,17 @@ pub fn parquet_to_arrow_schema(
     key_value_metadata: &Option<Vec<KeyValue>>,
 ) -> Result<Schema> {
     let mut metadata = parse_key_value_metadata(key_value_metadata).unwrap_or_default();
-    let arrow_schema_metadata = metadata
+    let maybe_schema = metadata
         .remove(super::ARROW_SCHEMA_META_KEY)
         .map(|encoded| get_arrow_schema_from_metadata(&encoded));
 
+    let arrow_schema_metadata = match maybe_schema {
```

Comment:
There may be an idiom for this in rust (that I do not know).

Reply:
@jorgecarleitao thank you for the tip!

The codes were updated to use `map_or` for flipping `Option<Result>` to `Result<Option>`.
BTW, I found some similar usage of `map_or` in `parquet/src/column/writer.rs` 🥇